common/lib/xmodule/xmodule/mongo_utils.py

```diff
@@ -33,7 +33,7 @@ def connect_to_mongodb(
     # The MongoReplicaSetClient class is deprecated in Mongo 3.x, in favor of using
     # the MongoClient class for all connections. Update/simplify this code when using
     # PyMongo 3.x.
-    if kwargs.get('replicaSet'):
+    if kwargs.pop('replicaSet', '') != '':
```

Comment:
Maxi, could you please make it more noob-friendly?
Please break the code into multiple lines so it's possible to understand the lines.

As far as I understand this is a hotfix, so we need to do the following:
```suggestion
    kwargs.pop('replicaSet', '')  # Hack: Compose.io expects no replicaSet` argument
    if False:  # TODO: Hack. Revert this check to the original: `if kwargs.get('replicaSet'):
````

Reply:
No, we need to live with this change Omar, I went for something more permanent.

Reply:
To be clear, I understand the urgency of this so I'll dismiss my own review so it's possible to get it merged.

That being said. I don't understand the change then as far as I know it removes `replicaSet`. Then we really need it to be noob-friendly because I cannot understand what it's supposed to do and why we chose this specific implementation.

Reply:
@OmarIthawi I added comments in the code, but maybe is still not clear. I'm gonna reword it here and then change them again the code comments if is more clear.

In order to connect to MongoDB Atlas, we need to provide the replica set name, we were lacking that variable in our configuration playbooks, so we add it. The problem is that we need a default there, an empty string for example. The problem, is that ansible YAML is really bad handling those defaults, our only option is to handle this in the code, but an empty it ends in an error, so we need to use None. 
That's why we're changing the empty string to None. Ideally, if isn't set, we should have a way in ansible, to avoid settings the replicaSet in the `lms.auth.json` and  `cms.auth.json`.